### PR TITLE
Update github action versions.

### PR DESCRIPTION
**Pull request by agalbachicar**
# 🎉 New feature

Part of https://github.com/maliput/maliput_infrastructure/issues/319

## Summary

Update github action versions.

## Test it
N/A

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by francocipollone — APPROVED** [review]

File: .github/workflows/manylinux_image_deploy.yml
Comment (francocipollone) [inline .github/workflows/manylinux_image_deploy.yml:1]: Should we try this workflow JIC?

Comment (agalbachicar) [reply .github/workflows/manylinux_image_deploy.yml:1]: https://github.com/maliput/maliput_infrastructure/actions/runs/9029129771

Comment (agalbachicar) [reply .github/workflows/manylinux_image_deploy.yml:1]: Fixed it in https://github.com/maliput/maliput_infrastructure/actions/runs/9029747091/job/24812681115

Comment (francocipollone) [reply .github/workflows/manylinux_image_deploy.yml:1]: Thanks for digging

**Comment by agalbachicar** [comment]
Gentle ping @francocipollone

File: wheel_generation/Dockerfile [hunk 1]
```diff
@@ -62,7 +62,7 @@ RUN echo "source /colcon_builds/install/setup.bash" >> /root/.bashrc
 ##########################################################################
 # Install vcstool
 ##########################################################################
-RUN /opt/_internal/cpython-3.8.18/bin/pip install vcstool
+RUN /opt/python/cp38-cp38/bin/pip install vcstool
```
Comment (francocipollone) [inline wheel_generation/Dockerfile:65]: Mind this:

 - Now that wheel generation is failing with the new generated image: See https://github.com/maliput/maliput_malidrive/actions/runs/9060260268/job/24889602445
 
 - Probably we need to change their use in maliput_malidrive/maliput_py's workflows as well: See https://github.com/maliput/maliput_malidrive/blob/main/.github/workflows/wheel_generation.yml#L18-L20

Comment (francocipollone) [reply wheel_generation/Dockerfile:65]: Solved at: https://github.com/maliput/maliput_malidrive/pull/282

**Review by francocipollone — APPROVED** [review]
LGTM

**Pull request merged**